lyrebird-vis/sdstreamer.cpp

```diff
@@ -79,39 +84,80 @@ int SDStreamer::configure_datavals(
             continue;
 
         cout << "config frame received, probably." << endl;
-        // Load visual elements.
+
+        // Vis Elem Info
+        G3VectorStringConstPtr n = frame->Get<G3VectorString>("cname");
         G3VectorDoubleConstPtr x = frame->Get<G3VectorDouble>("x");
         G3VectorDoubleConstPtr y = frame->Get<G3VectorDouble>("y");
         G3VectorDoubleConstPtr r = frame->Get<G3VectorDouble>("rotation");
         G3VectorStringConstPtr t = frame->Get<G3VectorString>("templates");
+
+        // Array of data values
+        G3VectorStringConstPtr dv = frame->Get<G3VectorString>("values");
+
+        // Equation info
         G3VectorStringConstPtr e = frame->Get<G3VectorString>("equations");
+        G3VectorStringConstPtr el = frame->Get<G3VectorString>("eq_labels");
         G3VectorStringConstPtr c = frame->Get<G3VectorString>("cmaps");
-        G3VectorStringConstPtr n = frame->Get<G3VectorString>("cname");
-        for (int i=0; i<(int)x->size(); i++) {
-            chan_names.push_back(n->at(i));
-            string eq_name = n->at(i) + "_equation";
+        G3VectorBoolConstPtr is_dynamic;
+        if (frame->Has("color_is_dynamic")) {
+            is_dynamic = frame->Get<G3VectorBool>("color_is_dynamic");
+        } else {
+            is_dynamic = boost::make_shared<G3VectorBool>(1, false);
+        }
+
+        int nchans = x->size();
+        if (e->size() % nchans > 0){
+            log_error("Equation array length must be a multiple of nchans");
+            break;
+        }
+        int eqs_per_chan = e->size() / n->size();
+
+        if (dv->size() % nchans > 0){
+            log_error("values array length must be a multiple of nchans");
+            break;
+        }
+        data_vals_per_chan = dv->size() / nchans;
+
+        // Setup data values
+        for (auto dv_name : *dv){
+                data_val_names.push_back(dv_name);
+        }
```

Comment:
indentation deviation ...

Also can you change "){" to ") {" in this and two blocks above?